### PR TITLE
change how times are handled in geojob -> webdata

### PR DESCRIPTION
**Pull request by wdwatkins**
Was only retaining dates

### CODE REVIEW

File: R/01-webdata-obj.R [hunk 1]
```diff
@@ -114,12 +114,16 @@ setMethod("webdata", signature("character"), function(.Object=c("prism",  "iclus
 #' @aliases webdata
 setMethod("webdata", signature("geojob"), function(.Object, ...) {
   xmlVals <- inputs(xmlParse(xml(.Object)))
+  browser()
```
Comment (jiwalker-usgs) [inline R/01-webdata-obj.R]: probably not good to check code debugging in

File: R/01-webdata-obj.R [hunk 2]
```diff
@@ -114,12 +114,16 @@ setMethod("webdata", signature("character"), function(.Object=c("prism",  "iclus
...
   url <- xmlVals[["DATASET_URI"]]
   times <- c(start = xmlVals[["TIME_START"]], end = xmlVals[["TIME_END"]])
   if(is.null(times[['start']])) {times[['start']] <- NA}
   if(length(times) == 1) {times[['end']] <- NA}
-  times <- as.POSIXct(c(times[['start']], times[['end']])) #could get out of order with one missing
-
+  times <- c(times[['start']], times[['end']])#could get out of order with one missing
+  times <- gsub(pattern = "T", replacement = " ", x = times)
+  times <- as.POSIXct(strftime(times,
```
Comment (jiwalker-usgs) [inline R/01-webdata-obj.R]: Can you just add the T to the format pattern?

File: R/01-webdata-obj.R (hunk 1 above)
Comment (wdwatkins) [inline R/01-webdata-obj.R]: 😬

File: R/01-webdata-obj.R (hunk 2 above)
Comment (wdwatkins) [inline R/01-webdata-obj.R]: I did try that, I'm not sure what I'm missing though:
```
> strftime('2011-12-25 23:00:00.000Z', format = "%Y-%m-%d %H:%M:%S", usetz = FALSE)
[1] "2011-12-25 23:00:00"
> strftime('2011-12-25T23:00:00.000Z', format = "%Y-%m-%dT%H:%M:%S", usetz = FALSE)
[1] "2011-12-25T00:00:00"```

Comment (wdwatkins) [inline R/01-webdata-obj.R]: With the T it loses the `23:00:00`

Comment (jiwalker-usgs) [inline R/01-webdata-obj.R]: hmm, fair enough

Comment (jread-usgs) [inline R/01-webdata-obj.R]: that is odd

Comment (jiwalker-usgs) [inline R/01-webdata-obj.R]: I think it is that `strftime` that is the problem here, the string is coerced to a POSIXct and then formatted in the output.  Instead this should be something like:
```
as.POSIXct(times, format="%Y-%m-%dT%H:%M:%S")
```

Comment (wdwatkins) [inline R/01-webdata-obj.R]: It works with the 'T' with `strptime`.

Comment (jiwalker-usgs) [inline R/01-webdata-obj.R]: strftime is for formatting output of a datetime object, strptime is for parsing and the timezones always suck

**Comment by jread-usgs** [comment]
is this all set, or pending more changes?

**Comment by wdwatkins** [comment]
Should be good to go, if @jiwalker-usgs is happy

**Pull request merged**